infra/roles/products/tasks/main.yml

```diff
@@ -0,0 +1,64 @@
+---
+- name: Install packages
+  apt:
+    name: "{{ item }}"
+    state: present
+  with_items:
+    - apache2
+    - openjdk-8-jre
+
+- name: Enable apache proxy modules
+  register: apache_mods
+  apache2_module:
+    state: present
+    name: "{{ item }}"
+  with_items:
+    - proxy
+    - proxy_http
+
+- name: Configure apache
+  register: apache_conf
+  copy:
+    src: apache_vhost
+    dest: /etc/apache2/sites-enabled/demo.conf
+
+- name: Restart apache
+  when: apache_conf.changed or apache_mods.changed
+  service:
+    name: apache2
+    state: restarted
+
+- name: Start apache
+  service:
+    name: apache2
+    state: started
+    enabled: yes
+
+- name: Upload systemd service file
+  copy:
+    src: JavaApp.service
+    dest: /etc/systemd/system/JavaApp.service
+    owner: root
+    group: root
+    mode: 0644
+
+- name: Upload restart script
+  copy:
+    src: restartApp.sh
+    dest: /home/ubuntu/restartApp.sh
```

Comment:
Are you sure you want to place the file in "ubuntu"s home when you are creating another user below.